### ISSUE TITLE
[Eloquent] Nullable fileds are saved as a string "null" when casting to array

### ISSUE THREAD

**Issue by Stichoza** [laravel/framework#8453]
### TL;DR

Nullable fields of type `JSON` are saved as a string `"null"` instead of `NULL` if casting to array.

### Description

I have a `JSON` type field which is also nullable. Created with migration like this:

```php
$table->json('cities')->nullable();
```

I'm casting this field to `array` like this:

```php
protected $casts = [
    'cities' => 'array'
];
```

There is no problem if I'm saving an array like `$object->cities = ['foo', 'bar']`. The problem occurs only when I set this field to `null` (as I mentioned, field is nullable).

`$object->cities = null` is saved as a string `"null"`. See screenshot from SequesPro (last two records):

![screen shot 2015-04-17 at 05 19 17](https://cloud.githubusercontent.com/assets/1139050/7194527/6b771ce2-e4c1-11e4-9ea1-cda19541eb2b.png)

## Solutions

I haven't found any solution, but if I disable casting, the same code saves data as `NULL` (as it should).

```php
protected $casts = [
    // 'cities' => 'array' <-- Saving as a NULL if casting is disabled
];
```

**Comment by Stichoza** [comment]
Maybe it's somehow linked to #7531, but why null as a string :fearful:
